internal/driver/glfw/window.go

```diff
@@ -446,19 +464,14 @@ func (w *window) Canvas() fyne.Canvas {
 }
 
 func (w *window) closed(viewport *glfw.Window) {
-	viewport.SetShouldClose(true)
+	viewport.SetShouldClose(false)
 
-	w.canvas.walkTrees(nil, func(node *renderCacheNode) {
-		switch co := node.obj.(type) {
-		case fyne.Widget:
-			cache.DestroyRenderer(co)
-		}
-	})
-
-	// trigger callbacks
-	if w.onClosed != nil {
-		w.queueEvent(w.onClosed)
+	if w.onCloseIntercepted != nil {
+		w.onCloseIntercepted()
```

Comment:
Earlier you mentioned that this could block - should it be passed to `w.queueEvent` instead?

Reply:
i've tried that, and it's not helping unfortunately
i can do this to follow the pattern

Reply:
Can you confirm what the issue is please?
Looking at the example above I can't help but feel the channel should not be needed - just calling `Window.Close()` in the callback should work.

Reply:
the issue is not reacting `ShowConfirm` 
Yes, it's solved with removing the channel with and without using `queueEvent`